lightning/src/ln/onion_utils.rs

```diff
@@ -1044,22 +1044,31 @@ where
 		let amt_to_forward = htlc_msat - route_hop.fee_msat;
 		htlc_msat = amt_to_forward;
 
-		let err_packet = match decrypt_onion_error_packet(&mut encrypted_packet, shared_secret) {
-			Ok(p) => p,
-			Err(_) => return,
-		};
+		let decrypt_result = decrypt_onion_error_packet(&mut encrypted_packet, shared_secret);
```

Comment:
Check hmac first to distinguish between unreadable failures and hmac mismatches.